apps/fido/fido.star

```diff
@@ -0,0 +1,265 @@
+"""
+Applet: Fido
+Summary: A pixel pal
+Description: Fido is a pixel pal that will sit, walk, and feast inside your Tidbyt.
+Author: yonodactyl
+"""
+
+# LOAD MODULES
+load("encoding/base64.star", "base64")
+load("render.star", "render")
+load("schema.star", "schema")
+load("http.star", "http")
+load("cache.star", "cache")
+load("encoding/json.star", "json")
+load("time.star", "time")
+load("math.star", "math")
+load("random.star", "random")
+load("humanize.star", "humanize")
+
+# CONFIG
+TTL = 86400
+DEFAULT_TIMEZONE = "America/New_York"
+DEFAULT_BIRTHDAY = "2022-12-29T10:00:00Z"
+LAST_DAY_WALKED = "2022-12-28T10:00:00Z"
+HOURS_IN_YEAR = 8760
+HOURS_IN_MONTH = 730
+DEFAULT_PAL_NAME = "Fido"
+
+# Static Base64 Image Data
+GRASS = base64.decode("""
+iVBORw0KGgoAAAANSUhEUgAAAEAAAAAgCAYAAACinX6EAAAAAXNSR0IArs4c6QAAAKJJREFUaIHt0CEOwyAUBuD/NRVcB2TlEmQVR+gROMeO0COgKudrtoxzzE8wxRRN2mymBsH/GfLzgDweQERERERERERERERE1AgBgGkdcwwJ2inMwyIAYLzN2qntYAwJz+tNpnXMJZf6PCxivM2lXvLZ+2ff/3X++L9jP/Kv2EruYkjbZotrVyahndpNppUsxtuMhvWXx7t2D1X199endg9VfQGuNRNssoJHlQAAAABJRU5ErkJggg==
+""")
+SKY = base64.decode("""
+iVBORw0KGgoAAAANSUhEUgAAAEAAAAAgCAYAAACinX6EAAAAAXNSR0IArs4c6QAAAXtJREFUaIHtVzsOwjAMpRU3YWHgWByFY3ApNhZ2DsBeFhIFp44/iZ2o6lseT0KRHT+/ttPl/VkOnfC8vw7n6ynTnjxLiuVoCYcitM1T52A65cnCAdhkoe7BAagDWkyaO5GaJqSTxvowcQCGkZxAZsBWd37PgB+iAzwnrW2eey6HYV+mGcB1AtSeTjDNAO3u1jqDwwF7BoyYAdxzNAz7GiIDekw+YE6LgcWvaU8nWOw81PEC0smU9Bqnky3pWsbOLzHVz3S8PZbeAZUWVdIWmCWTljLXCdT/W9f1dwHhh2cG1O58y0xQZUDvnS9pbj/xAjSTbp3u2smvMewD0wFN3wPgJDA9UtCSGdBi57nNS8+R1If1SWaAJN2h9th5LN2xnYeYpTdpkQHcczkcgO18dgE1z1Pt5LSTl3AApasyQLu7tc7gMNVXdgEjZADXGSWm+oGI3wIejyLqkQS1B1wywGPntU6IL0KeDpA6wxJZBmx95yFcMiAtqvfOQ3wBeCUrKD8qqeQAAAAASUVORK5CYII=
+""")
+FENCE = base64.decode("""
+iVBORw0KGgoAAAANSUhEUgAAAEAAAAAgCAYAAACinX6EAAAAAXNSR0IArs4c6QAAAUdJREFUaIHtlTFLAzEYht/EA5cDl8IpXUQcOpaCo3Tp4Og/8S/4F/wnjg5ZCt0OSsYudhQzCha8QeqUMx6XS9LkWqHfs1yekLt78+U7DiAIgiAIgiAIgiCODwYALw/3Wz1x9/TMfFxIhdm4+Lfuux8+HU1qAQBfn42LaBdSBXnI8333w/VASGXO78V1aCFVp/eZ5+RycPFYfX7VL1u8fsDmQiq8bb6T+zDPOj32+bfXZ7g6z1udA7/tprG5LorNXffb3PfEmh3hymNbb/qfDoit+Pp9s5MP86wOZvO+OoR3VSj0hGLcRcr3mR3I2ya1m4S6i1JVrWNfYvKZh93bX6CLUlWYr5ZMb9wc+5Iqb4YDMF8tmXk9JNy9JB365Jvz09Fke1Oc7jNKjbMDUgdrbnaX7z8EV34WWv1SVVFFCb2/7/UEQRBHzQ8p2EzduEqDNAAAAABJRU5ErkJggg==
+""")
+
+# Animated Actions
+BALL_THROW = "https://raw.githubusercontent.com/Yonodactyl/TidbytGIFs/main/Fido/ball_throw.gif"
```

Comment:
Are these mostly static or meant to change over time? If static, perhaps it'd be better to just embed their base64 encoding in this script directly? Could make it easier for other devs to suggest changes, and eliminates some HTTP calls as well.

Reply:
For now - they are static. My goal is to allow these to be modifiable later on. I went with the link to have less code mostly.

I think removing the HTTP calls would be great though. To get over the hump of increasing LOC count, I can probably add the base64 at the end!

Reply:
Tips to keep the size of the images down:
- Scale down the file to whatever dimensions you're using in the app (e.g. `convert -scale 32x32 large.png small.png` if you've imagemagick installed)
- Strip exif data from the file (e.g. `exiftool -all= small.png`)